libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/LayoutSyncManager.java

```diff
@@ -0,0 +1,288 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.smartsync.manager;
+
+import com.salesforce.androidsdk.accounts.UserAccount;
+import com.salesforce.androidsdk.app.SalesforceSDKManager;
+import com.salesforce.androidsdk.smartstore.store.IndexSpec;
+import com.salesforce.androidsdk.smartstore.store.QuerySpec;
+import com.salesforce.androidsdk.smartstore.store.SmartStore;
+import com.salesforce.androidsdk.smartsync.app.SmartSyncSDKManager;
+import com.salesforce.androidsdk.smartsync.model.Layout;
+import com.salesforce.androidsdk.smartsync.target.LayoutSyncDownTarget;
+import com.salesforce.androidsdk.smartsync.target.SyncDownTarget;
+import com.salesforce.androidsdk.smartsync.util.SmartSyncLogger;
+import com.salesforce.androidsdk.smartsync.util.SyncOptions;
+import com.salesforce.androidsdk.smartsync.util.SyncState;
+
+import org.json.JSONArray;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides an easy way to fetch layout data using {@link com.salesforce.androidsdk.smartsync.target.LayoutSyncDownTarget}.
+ * This class handles creating a soup, storing synched data and reading it into
+ * a meaningful data structure, i.e. {@link com.salesforce.androidsdk.smartsync.model.Layout}.
+ *
+ * @author bhariharan
+ */
+public class LayoutSyncManager {
+
+    private static final String SOUP_NAME = "layouts";
+    private static final String FEATURE_LAYOUT_SYNC = "LY";
+    private static final String QUERY = "SELECT {" + SOUP_NAME + ":_soup} from {" + SOUP_NAME +
+            "} WHERE {" + SOUP_NAME + ":objectType} = '%s' and {" + SOUP_NAME + ":layoutType} = '%s'";
+    private static final String TAG = "LayoutSyncManager";
+    private static final IndexSpec[] INDEX_SPECS = new IndexSpec[] {
+        new IndexSpec("objectType", SmartStore.Type.string),
```

Comment:
Should we use type json1? Maybe at some point in the future we will deprecate other index types - or maybe we will need to add indexes to that table, when using json1, index changes can be done very quickly (because it's pure sql).

Reply:
Okay - I can take a look at that.